internal/dataplane/parser/translate_failures_test.go

```diff
@@ -14,11 +14,17 @@ import (
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/store"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/versions"
 	kongv1beta1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1beta1"
 )
 
 // This file contains unit test functions to test translation failures genreated by parser.
 
+func newResourceFailure(reason string, objects ...client.Object) failures.ResourceFailure {
+	failure, _ := failures.NewResourceFailure(reason, objects...)
```

Comment:
How are we sure that the error won't be returned and we just miss it?

Reply:
Used `require` to fail the test if error happens.